src/controllers/v1/members.ts

```diff
@@ -59,19 +69,31 @@ export class MembersController extends Controller {
     if (!nodeMembers.includes(aliasOrAddress)) {
       throw new NotFound('Member does not exist')
     }
+    if (aliasOrAddress === this.env.get('SELF_ADDRESS')) {
+      return {
+        alias: aliasOrAddress,
+        address: aliasOrAddress,
+        role: 'Self',
+      }
+    }
 
     return {
       alias: aliasOrAddress,
       address: aliasOrAddress,
+      role: 'None',
     }
   }
 
   @SuccessResponse(200)
   @Response<NotFound>(404, 'Not found')
   @Response<Conflict>(409, 'Conflict')
   @Put('/{address}')
-  public async put(@Path('address') address: Address, @Body() body: { alias: Alias }): Promise<Member> {
+  public async put(
+    @Path('address') address: Address,
+    @Body() body: { alias: Alias; role?: 'None' | 'Optimiser' }
+  ): Promise<Member> {
     const { alias } = body
+    const role = body.role ?? 'None'
```

Comment:
This will mean if we don't specify a role it will still be set to `None`. Is that what we want?